bool healthy
uint8 behaviour
uint8 speed
